cli/internal/cmd/upgradecheck.go

```diff
@@ -661,16 +717,20 @@ func (v *versionCollector) filterCompatibleCLIVersions(ctx context.Context, cliP
 }
 
 type upgradeCheckFlags struct {
-	configPath  string
-	force       bool
-	writeConfig bool
-	ref         string
-	stream      string
+	configPath        string
+	force             bool
+	writeConfig       bool
+	ref               string
+	stream            string
+	terraformLogLevel terraform.LogLevel
 }
 
 type upgradeChecker interface {
```

Comment:
the interface seems to get more away from what the name suggests: "checking" and instead has side-effects. 
PlanTerraformMigrations and CleanUpTerraformMigrations don't seem to belong there.
By seperating this we could maybe also avoid passing the same checker object twice in `runUpgradeCheck` for cmd creation?